test/Infinispan.HotRod.Tests/RemoteCacheAdminTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Infinispan.HotRod.Config;
+using Infinispan.HotRod.Tests.Util;
+using Infinispan.HotRod.Transport;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace Infinispan.HotRod.Tests.ClusteredXml2
+{
+    [TestFixture]
+    [Category("clustered_xml_2")]
+    [Category("ClusterTestSuite")]
+    public class RemoteCacheAdminTest
+    {
+        RemoteCacheManager remoteManager;
+        IMarshaller marshaller;
+
+        [SetUp]
+        public void BeforeAnyTest()
+        {
+            ClusterTestSuite.EnsureServersUp();
+            remoteManager = ClusterTestSuite.getRemoteCacheManager();
+        }
+
+        [TearDown]
+        public void AfterAnyTest()
+        {
+            remoteManager.Stop();
+        }
+
+        [Test]
+        public void alreadyExistingCacheTest()
+        {
+            remoteManager.Administration().CreateCache<string, string>("alreadyExistingCache", "template");
+            var ex = Assert.Throws<Infinispan.HotRod.Exceptions.HotRodClientException>(() =>
+                { remoteManager.Administration().CreateCache<string, string>("alreadyExistingCache", "template"); });
+            StringAssert.Contains("ISPN000507:", ex.Message);
+        }
+
+        [Test]
+        public void nonExistentTemplateTest()
+        {
+            var ex = Assert.Throws<Infinispan.HotRod.Exceptions.HotRodClientException>(() =>
+            { remoteManager.Administration().CreateCache<string, string>("cache4AdminTest", "nonExistentTemplate"); });
+            StringAssert.Contains("ISPN000374:", ex.Message);
+        }
+
+        [Test]
+        public void getOrCreateWithoutTemplateTest()
+        {
+            remoteManager.Administration().GetOrCreateCache("default", (String)null);
```

Comment:
I'm missing some assertions here. Perhaps something like in cacheCreateWithXMLConfigurationAndGetCacheTest

Reply:
ok